src/App.js

```diff
@@ -13,15 +23,33 @@ const welcomeInfo = {
   userName: "Mrs Flanders",
   workbooks: ["Dinosaurs", "Dinosaurs Eggs", "Dinosaur Train", "Dino Run!"],
   students: ["Mickey", "Rickey", "Vinny", "the", "Poo"]
-
 }
 
 function App() {
+  const messageService = useRef(messagingService);
+  const [loadingState, setLoadingState] = useState(true);
+
+  useEffect(() => {
+    const connectToBroker = async () => {
+      await messageService.current.connect();
+      setLoadingState(false);
+    }
+    const cleanup = () => messageService.current.disconnect();
+    connectToBroker();
+    return cleanup;
+  }, [messageService]);
+
   const students = {
     totalStudentsHappy: 1,
     totalStudentsNeutral: 15,
     totalStudentsSad: 23,
   };
+  return loadingState 
+  ? (<div>loading messaging service...</div>) 
+  : (
+    <div className="App">
+      <MessagingServiceContext.Provider value={messagingService}>
```

Comment:
The indentation is throwing me off here. Is everything a child of this messaging div?

Reply:
Yea, it's more of a React Context provider component than a div.